src/branchrule.rs

```diff
@@ -73,16 +73,17 @@ mod tests {
 
     #[test]
     fn choosing_first_branching_rule() {
-        let br = FirstChoosingBranchingRule { chosen: None };
+        let mut br = FirstChoosingBranchingRule { chosen: None };
 
         let model = Model::new()
             .set_longint_param("limits/nodes", 2) // only call brancher once
             .unwrap()
             .hide_output()
             .include_default_plugins()
             .read_prob("data/test/gen-ip054.mps")
-            .unwrap()
-            .include_branch_rule("", "", 100000, 1000, 1., Box::new(br));
+            .unwrap();
+
+        model.include_branch_rule("", "", 100000, 1000, 1., &mut br);
 
```

Comment:
I'm not really sure but wouldn't drop(br) here compile but would result in a use after free during the solve?